lib/ecto/changeset.ex

```diff
@@ -1842,9 +1850,9 @@ defmodule Ecto.Changeset do
   defp no_assoc_message(:one), do: "is still associated to this entry"
   defp no_assoc_message(:many), do: "are still associated to this entry"
 
-  defp add_constraint(changeset, type, constraint, field, error)
-       when is_binary(constraint) and is_atom(field) and is_tuple(error) do
-    update_in changeset.constraints, &[%{type: type, constraint: constraint,
+  defp add_constraint(changeset, type, constraint, match \\ :exact, field, error)
```

Comment:
It would be best if we didn't use the default argument so we are always forced to explicitly pass the option.